### PR TITLE
refactor(tests): move all tests from mocha to jest (huge migration)

### PR DESCRIPTION
**Pull request by marpme**
# First iteration
e3d7a74 -> 2812c86

**Removes:**

 * Webpack 2 
 * Mocha 6
 * NYC / Istanbul
 * Enzyme
 * Chai (+ plugins)
 * XUnit

(all of the above where solely being used for test purposes)

**and replaces it with:**

* Jest 
* Jest-Juint

Tests were [migrated](https://jestjs.io/docs/migration-guide) via `jest-codemod` from mocha & chai to jest expectations.
Adjusting package.json and respective circleCI test execution.

# Secondary Iteration

30459e6 -> 9dfd571

Bumps all babel dependencies and fixes them on the latest version
And fixes some flow check errors, which occur only on the latest babel version 🤷🏻 

**Result:**

image

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * https://jestjs.io/docs/configuration
+ */
+
+module.exports = {
+  clearMocks: true,
+  collectCoverage: true,
+  coverageDirectory: "coverage",
+  coveragePathIgnorePatterns: [
+    "/node_modules/"
+  ],
```
Comment (Matwog) [inline jest.config.js:12]: **question:** Just as a general question for these config files, I see that this flag takes `["/node_modules/"]` as default value. Now for flags which have our expected value as default value, like this case, should we still explicitly add this flag?

File: package.json [hunk 2]
```diff
@@ -41,23 +41,23 @@
...
+    "@babel/plugin-proposal-numeric-separator": "^7.14.5",
+    "@babel/plugin-proposal-throw-expressions": "^7.14.5",
+    "@babel/plugin-syntax-dynamic-import": "^7.8.3",
+    "@babel/plugin-syntax-import-meta": "^7.10.4",
+    "@babel/plugin-transform-proto-to-assign": "^7.14.5",
+    "@babel/plugin-transform-runtime": "^7.14.5",
+    "@babel/polyfill": "^7.12.1",
+    "@babel/preset-env": "^7.14.7",
+    "@babel/preset-flow": "^7.14.5",
+    "@babel/preset-react": "^7.14.5",
```
Comment (Matwog) [inline package.json:60]: The following plugins are already included in `@babel/preset-env`

- `@babel/plugin-proposal-class-properties`
- `@babel/plugin-proposal-export-namespace-from`
-  `@babel/plugin-proposal-json-string`
-  `@babel/plugin-proposal-numeric-separator`

So would it make sense for us to remove them from the `devDependencies`?

File: package.json [hunk 3]
```diff
@@ -41,23 +41,23 @@
...
+    "@babel/plugin-proposal-export-namespace-from": "^7.14.5",
+    "@babel/plugin-proposal-function-sent": "^7.14.5",
+    "@babel/plugin-proposal-json-strings": "^7.14.5",
+    "@babel/plugin-proposal-numeric-separator": "^7.14.5",
+    "@babel/plugin-proposal-throw-expressions": "^7.14.5",
+    "@babel/plugin-syntax-dynamic-import": "^7.8.3",
+    "@babel/plugin-syntax-import-meta": "^7.10.4",
+    "@babel/plugin-transform-proto-to-assign": "^7.14.5",
+    "@babel/plugin-transform-runtime": "^7.14.5",
+    "@babel/polyfill": "^7.12.1",
```
Comment (Matwog) [inline package.json:57]: **question:** I believe `@babel/polyfill` is deprecated. 
🚨 As of Babel 7.4.0, this package has been deprecated in favor of directly including core-js/stable (to polyfill ECMAScript features) and regenerator-runtime/runtime (needed to use transpiled generator functions)

**Review by Matwog — COMMENTED** [review]
Solid work 🎉 
Left few comments.

File: jest.config.js (hunk 1 above)
Comment (marpme) [reply jest.config.js]: 0400d0e

File: package.json (hunk 2 above)
Comment (marpme) [reply package.json]: 122fa94

File: package.json (hunk 3 above)
Comment (marpme) [reply package.json]: d7d16d4 - checked the build output and it is really unused. wow.

File: jest.config.js (hunk 1 above)
Comment (marpme) [reply jest.config.js]: Generally not, I didn't know that this is also the default value :o Thanks for hinting that!

**Review by Matwog — APPROVED** [review]
Reviewed the latest changes. LGTM 🎉

**Pull request merged**